documentation/src/main/asciidoc/public/reference/_static-metamodel-overview.adoc

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Red Hat Inc. and Hibernate Authors
+[[static-metamodel-overview]]
+= Overview of the static metamodel
```

Comment:
@yrodiere this commit (https://github.com/hibernate/hibernate-search/pull/4156/commits/24465fae98c562209953d8fc5d2d4995ccdc575e) adds the docs 😃. only pinging you here so it'll be easier to find this change 😃 🙈